### PR TITLE
[bitnami/kiam] MAJOR - Update Kiam to v4 + agent root permissions

### PR DESCRIPTION
**Pull request by migruiz4**
This PR follows up #5052.

In addition to updating Kiam to version 4, updates the agent permissions to run as root + NET_ADMIN capabilities.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by FraPazGal — COMMENTED** [review]
Pointed out a minor issue with missing values. The rest of the standardization changes and the ones needed for kiam@4 look good.

I think we can close #5052, any cons on this @joancafom?

File: bitnami/kiam/templates/agent/agent-daemonset.yaml [hunk 1]
```diff
@@ -147,6 +153,18 @@ spec:
...
+            tcpSocket:
+              port: {{ .Values.agent.containerPort }}
+            initialDelaySeconds: {{ .Values.agent.startupProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.agent.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.agent.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.agent.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.agent.startupProbe.failureThreshold }}
+          {{- else if .Values.agent.customStartupProbe }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.agent.customStartupProbe "context" $) | nindent 12 }}
+          {{- end }}
```
Comment (juan131) [inline bitnami/kiam/templates/agent/agent-daemonset.yaml:167]: I'm missing support for "diagnostic mode" here.

File: bitnami/kiam/templates/server/server-daemonset.yaml [hunk 2]
```diff
@@ -134,6 +140,18 @@ spec:
...
+            grpc:
+              port: grpclb
+            initialDelaySeconds: {{ .Values.server.startupProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.server.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.server.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.server.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.server.startupProbe.failureThreshold }}
+          {{- else if .Values.server.customStartupProbe }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.server.customStartupProbe "context" $) | nindent 12 }}
+          {{- end }}
```
Comment (juan131) [inline bitnami/kiam/templates/server/server-daemonset.yaml:154]: Same here: diagnostic mode.

File: bitnami/kiam/templates/server/server-deployment.yaml [hunk 3]
```diff
@@ -135,6 +141,18 @@ spec:
...
+            grpc:
+              port: grpclb
+            initialDelaySeconds: {{ .Values.server.startupProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.server.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.server.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.server.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.server.startupProbe.failureThreshold }}
+          {{- else if .Values.server.customStartupProbe }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.server.customStartupProbe "context" $) | nindent 12 }}
+          {{- end }}
```
Comment (juan131) [inline bitnami/kiam/templates/server/server-deployment.yaml:155]: And here again

**Review by juan131 — CHANGES_REQUESTED** [review]
Please add support for diagnostic mode in KIAM containers.

**Comment by joancafom** [comment]
> I think we can close #5052, any cons on this @joancafom?

No, I will go ahead and close that PR in favor of this one.

**Comment by migruiz4** [comment]
@juan131  @FraPazGal I updated this PR with the suggestions, please re-review when possible.

**Review by juan131 — APPROVED** [review]
LGTM

**Review by FraPazGal — DISMISSED** [review]
LGTM

File: bitnami/kiam/templates/agent/agent-service.yaml [hunk 4]
```diff
@@ -43,6 +49,9 @@ spec:
       {{- else if eq .Values.agent.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+    {{- if .Values.service.agent.extraPorts }}
```
Comment (voor) [inline bitnami/kiam/templates/agent/agent-service.yaml:52]: These two values are reversed, it should be `.Values.agent.service.extraPorts`

Comment (migruiz4) [reply bitnami/kiam/templates/agent/agent-service.yaml:52]: Thank you @voor, I have fixed this in this PR: https://github.com/bitnami/charts/pull/10130

**Pull request merged**